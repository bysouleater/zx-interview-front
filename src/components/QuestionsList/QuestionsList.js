import React from 'react';
import PropTypes from 'prop-types';
import QuestionCard from '../QuestionCard/QuestionCard';

const QuestionsList = ({ questions, setAnswerValue }) => (
  <div>
    {questions.map((question) => (
      <QuestionCard
        key={question.id}
        question={question}
        setAnswerValue={setAnswerValue}
      />
    ))}
  </div>
);

QuestionsList.propTypes = {
  questions: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
  })),
  setAnswerValue: PropTypes.func.isRequired,
};

QuestionsList.defaultProps = {
  questions: [],
};

export default QuestionsList;
