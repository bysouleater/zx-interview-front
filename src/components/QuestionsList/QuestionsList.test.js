import React from 'react';
import { render } from '@testing-library/react';
import QuestionsList from './QuestionsList';

test('renders empty list', () => {
  const { container } = render(<QuestionsList setAnswerValue={() => {}} />);
  expect(container).toMatchSnapshot();
});

test('renders questions', () => {
  const questions = [{ id: 'test1' }, { id: 'test2' }];
  const { container } = render(<QuestionsList setAnswerValue={() => {}} questions={questions} />);
  expect(container).toMatchSnapshot();
});
