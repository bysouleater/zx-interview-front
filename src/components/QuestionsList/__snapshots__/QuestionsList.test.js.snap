// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty list 1`] = `
<div>
  <div />
</div>
`;

exports[`renders questions 1`] = `
<div>
  <div>
    <div
      class="ui segment"
    >
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <label
            class="sc-AxjAm iLMcqR"
          />
          <div
            class="field sc-AxirZ gvygHu"
          >
            <div
              class="ui input"
            >
              <input
                placeholder="Your answer"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="ui segment"
    >
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <label
            class="sc-AxjAm iLMcqR"
          />
          <div
            class="field sc-AxirZ gvygHu"
          >
            <div
              class="ui input"
            >
              <input
                placeholder="Your answer"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
